### ISSUE TITLE
Runtime `require()` gives "file not found" error even when file exists

### ISSUE THREAD

**Issue by jahooma** [oven-sh/bun#14147]
### What version of Bun is running?

1.1.29+6d43b3662

### What platform is your computer?

Darwin 23.5.0 arm64 arm

### What steps can reproduce the bug?

I'm using workspaces. One workspace is called "test" and it imports code from the "common" workspace.

When I run `bun test` in the "test" workspace, it gives the following error for one of my test files:

```
error: ENOENT: No such file or directory
      at load (/Users/jahooma/manicode/node_modules/node-gyp-build/node-gyp-build.js:26:34)
      at /Users/jahooma/manicode/node_modules/tree-sitter/index.js:1:7
```

The "common" workspace uses the tree-sitter package, which uses the node-gyp-build package.

Bun has stored these packages at the root node_modules of my project.

In the line that errors, node-gyp-build is calling:

```js
require("/Users/jahooma/manicode/node_modules/tree-sitter/prebuilds/darwin-arm64/tree-sitter.node")
```

This file totally exists on my machine, but throws an error. I don't think I'm doing anything special here.t

### What is the expected behavior?

Require should load the file and not throw an error

### What do you see instead?

```bash
> test git:(main) ✗ bun run test-manifold
$ bun test manifold-project.test.ts
bun test v1.1.29 (6d43b366)

__src__/manifold-project.test.ts:
Loading native module /Users/jahooma/manicode/node_modules/tree-sitter
Resolved path /Users/jahooma/manicode/node_modules/tree-sitter/prebuilds/darwin-arm64/tree-sitter.node
file exists true
file loads 406864

# Unhandled error between tests
-------------------------------
21 | function load (dir) {
22 |   console.log('Loading native module', dir)
23 |   console.log('Resolved path', load.resolve(dir))
24 |   console.log('file exists', fs.existsSync(load.resolve(dir)))
25 |   console.log('file loads', fs.readFileSync(load.resolve(dir)).length)
26 |   console.log('Runtime require', require(load.resolve(dir)))
                                      ^
error: ENOENT: No such file or directory
      at load (/Users/jahooma/manicode/node_modules/node-gyp-build/node-gyp-build.js:26:34)
      at /Users/jahooma/manicode/node_modules/tree-sitter/index.js:1:7
-------------------------------


 0 pass
 1 fail
 1 error
Ran 1 tests across 1 files. [242.00ms]
error: script "test-manifold" exited with code 1
```

**Comment by jahooma** [comment]
I tried using tree-sitter outside a test and it linked me to the napi issue:
https://github.com/oven-sh/bun/issues/158

Seems like this was the problem here. Would have been nice if it threw an appropriate error, because I spent a lot of time trying to figure out why it could not read that file.

**Comment by Jarred-Sumner** [comment]
Fixed in #14915

**Comment by jahooma** [comment]
Hooray!!
